airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -107,12 +108,26 @@ def validate_and_fill_with_defaults(config: MutableMapping[str, Any]) -> Mutable
         return config
 
     def check_connection(self, logger: logging.Logger, config: MutableMapping[str, Any]) -> Tuple[bool, Any]:
-        self.validate_and_fill_with_defaults(config)
-        stripe.api_key = config["client_secret"]
+        config = self.validate_and_fill_with_defaults(config)
```

Comment:
nit: the logic to prepare the config seems very similar than the one in `streams`. Should we extract that?